### DISCUSSION TITLE
Problem with tmp node modules: Error: Cannot find module 'tmp'

### DISCUSSION THREAD

**Discussion by Pabluskagamo** [deta/cloud-docs#612] [Help]
Hello everyone, i recently deployed mi api to deta-micros and i got a problem with one of the modules im using. This module has a dependency with the module tmp which is used to create temporary files. The problem is that this module is not found and im getting the following error on deta visor:

Error: Cannot find module 'tmp'

Since it was a dependency of another module i didnt have tmp module on package.json so i tried to install it and deploy again but i got the same error. 

I hope you can help me and thank you for your atention.

**Comment by abdelhai** [comment]
hi, please provide code samples, links etc to make it easier to help you. also make sure to test locally first.

**Reply by Pabluskagamo** [reply]
Hi again, I tried everything locally and it works fine. The problem is with a node module, ExcelJS which im using to generate excel files. As i said it works locally fine but when i deploy to deta i got the following error from deta visor: 

Error: Cannot find module 'tmp'
Require stack:
- /opt/nodejs/node_modules/exceljs/lib/stream/xlsx/workbook-reader.js
- /opt/nodejs/node_modules/exceljs/lib/exceljs.nodejs.js
- /opt/nodejs/node_modules/exceljs/excel.js
- /var/task/src/services/excelCreator/excelCreator.js

this happens when i import the module:
const ExcelJS = require('exceljs');

i was thinking the problem is maybe that node is getting confused with tmp folder but im not sure.

I hope you can help me. Thank you for your atention.

**Reply by Kirlovon** [reply]
Hi! I did some reproduction of this, and seems like it happens specifically with the `tmp` module. For some reason it is not added to the `node_modules` folder, regardless of whether it is added to `package.json` or is a dependency of one of the dependencies. Module itself is fine, it is something with the dependencies installation.

As a temporary fix, you can copy `tmp` folder from `node_modules` to your project root directory (so that it is in the same directory as the package.json), so it will be deployed with your code. I'm not sure exceljs will work ok with this hack, but the import error will definitely disappear.

**Reply by Pabluskagamo** [reply]
Hello! Thank you so much for your time, i tried that option and now the deployment is working but im not sure if the module is working properly i will be testing to make it work. If you discover the real problem why tmp module is not getting installed notice me. 

Thank you so much for your atention and time.
